common/src/main/java/com/wynntils/features/combat/HorseMountFeature.java

```diff
@@ -119,11 +118,43 @@ private void mountHorse(Entity horse) {
 
         // swap to soul points to avoid right click problems
         int prevItem = McUtils.inventory().selected;
-        McUtils.sendPacket(new ServerboundSetCarriedItemPacket(InventoryUtils.SOUL_POINTS_SLOT_NUM));
+        int nonConflictingSlot = findNonConflictingSlot();
+        if (nonConflictingSlot == -1) {
+            postHorseErrorMessage(MountHorseStatus.CONFLICTING_SLOTS);
+            return;
+        }
+
+        McUtils.sendPacket(new ServerboundSetCarriedItemPacket(nonConflictingSlot));
         McUtils.sendPacket(ServerboundInteractPacket.createInteractionPacket(horse, false, InteractionHand.MAIN_HAND));
         McUtils.sendPacket(new ServerboundSetCarriedItemPacket(prevItem));
     }
 
+    /** Finds a hotbar slot where the item held allows us to safely mount a horse */
+    private int findNonConflictingSlot() {
+        for (int i = 0; i < 9; i++) {
+            ItemStack itemStack = McUtils.inventory().getItem(i);
+
+            // empty hand
+            if (itemStack.isEmpty()) {
+                return i;
+            }
+
+            // horse item
+            Optional<HorseItem> horseItemOpt = Models.Item.asWynnItem(itemStack, HorseItem.class);
+            if (horseItemOpt.isPresent()) {
+                return i;
+            }
+
+            // weapons (temp. disabled due to wynn bug)
+            //            Optional<GearItem> gearItemOpt = Models.Item.asWynnItem(itemStack, GearItem.class);
+            //            if (gearItemOpt.isPresent() && gearItemOpt.get().getGearType().isWeapon()) {
+            //                return i;
+            //            }
```

Comment:
I think, just enable this, and write it off as a wynn bug. They'll fix it, and we won't have to do a new patch.